src/models/lineChart.js

```diff
@@ -383,10 +383,17 @@ nv.models.lineChart = function() {
                 var singlePoint, pointIndex, pointXLocation, allData = [];
                 data
                     .filter(function(series, i) {
-                        series.seriesIndex = i;
-                        return !series.disabled && !series.disableTooltip;
+                        // Assign a color index to the series if it's not disabled
+                        // to use for the tooltip colors rather than the series index
+                        if(!series.disabled && !series.disableTooltip) {
+                            series.colorIndex = i;
+                            return series
+                        }
                     })
                     .forEach(function(series,i) {
+                        // Assign the index here instead of in the filter
+                        // as the DOM removes disabled series
+                        series.seriesIndex = i;
```

Comment:
Isn't this going to make two different series have the same seriesIndex if you start off with some disabled and enable things since this is after it's been filtered?    I think this should still be in the .filter(...) but before the if block right?